src/function.rs

```diff
@@ -0,0 +1,465 @@
+use core::{cmp::Ordering, iter};
+
+use alloc::{borrow::Cow, boxed::Box};
+
+use crate::{Error, RangeAttributes, ResDwarf, demangle_auto, lazy::LazyCell, maybe_small::{self, Vec}, name_attr};
```

Comment:
Please don't use nested imports because they make it too hard to see the module dependencies. (eg lazy::LazyCell should be on a separate line).

Reply:
`name_attr` should be moved to this module too.

Reply:
if we move `name_attr`, we should also move `name_entry` as it is only use in `name_attr`